From 798ae737a1a72fec487b2cccb5a045bd52514d00 Mon Sep 17 00:00:00 2001
From: Christian Stewart <christian@paral.in>
Date: Thu, 14 Apr 2022 19:38:54 -0700
Subject: [PATCH] sunxi-common: increase kernel size for risc-v

This riscV machine (Nezha) has 1GB RAM similar to a ARM64 machine, not the
maximum of 8MB uncompressed kernel specified in the header.

Copy the ARM64 defines to the RISC-V section to allow a larger kernel.

Signed-off-by: Christian Stewart <christian@paral.in>
---
 include/configs/sunxi-common.h | 34 +++++++++++-----------------------
 1 file changed, 11 insertions(+), 23 deletions(-)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index 7fc0fe63f0..9c85b255b3 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -203,22 +203,18 @@
 #define RAMDISK_ADDR_R    __stringify(SDRAM_OFFSET(FF00000))
 
 #elif defined(CONFIG_MACH_SUN8I_V3S) || defined(CONFIG_RISCV)
-/*
- * 64M RAM minus 2MB heap + 16MB for u-boot, stack, fb, etc.
- * 16M uncompressed kernel, 8M compressed kernel, 1M fdt,
- * 1M script, 1M pxe, 1M dt overlay and the ramdisk at the end.
- */
-#define BOOTM_SIZE        __stringify(0x2e00000)
-#ifdef CONFIG_RISCV
+
+// SkiffOS: copy settings from arm64
+#define BOOTM_SIZE        __stringify(0xa000000)
+// #define KERNEL_ADDR_R     __stringify(SDRAM_OFFSET(0080000))
 #define KERNEL_ADDR_R     __stringify(SDRAM_OFFSET(0040000))
-#else
-#define KERNEL_ADDR_R     __stringify(SDRAM_OFFSET(1000000))
-#endif
-#define FDT_ADDR_R        __stringify(SDRAM_OFFSET(1800000))
-#define SCRIPT_ADDR_R     __stringify(SDRAM_OFFSET(1900000))
-#define PXEFILE_ADDR_R    __stringify(SDRAM_OFFSET(1A00000))
-#define FDTOVERLAY_ADDR_R __stringify(SDRAM_OFFSET(1B00000))
-#define RAMDISK_ADDR_R    __stringify(SDRAM_OFFSET(1C00000))
+#define KERNEL_COMP_ADDR_R __stringify(SDRAM_OFFSET(4000000))
+#define KERNEL_COMP_SIZE  __stringify(0xb000000)
+#define FDT_ADDR_R        __stringify(SDRAM_OFFSET(FA00000))
+#define SCRIPT_ADDR_R     __stringify(SDRAM_OFFSET(FC00000))
+#define PXEFILE_ADDR_R    __stringify(SDRAM_OFFSET(FD00000))
+#define FDTOVERLAY_ADDR_R __stringify(SDRAM_OFFSET(FE00000))
+#define RAMDISK_ADDR_R    __stringify(SDRAM_OFFSET(FF00000))
 
 #elif defined(CONFIG_MACH_SUNIV)
 /*
@@ -258,18 +254,10 @@
 	"fdtoverlay_addr_r=" FDTOVERLAY_ADDR_R "\0" \
 	"ramdisk_addr_r=" RAMDISK_ADDR_R "\0"
 
-#ifdef CONFIG_ARM64
-
 #define MEM_LAYOUT_ENV_EXTRA_SETTINGS \
 	"kernel_comp_addr_r=" KERNEL_COMP_ADDR_R "\0" \
 	"kernel_comp_size=" KERNEL_COMP_SIZE "\0"
 
-#else
-
-#define MEM_LAYOUT_ENV_EXTRA_SETTINGS ""
-
-#endif
-
 #define DFU_ALT_INFO_RAM \
 	"dfu_alt_info_ram=" \
 	"kernel ram " KERNEL_ADDR_R " 0x1000000;" \
-- 
2.35.1

